assets/js/common/CustomCheckModal/CustomCheckModal.jsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import { noop } from 'lodash';
+import { EOS_WARNING_OUTLINED } from 'eos-icons-react';
+
+import Modal from '@common/Modal';
+import Button from '@common/Button';
+import Input from '@common/Input';
+import Label from '@common/Label';
+import ProviderLabel from '@common/ProviderLabel';
+import Tooltip from '@common/Tooltip';
+
+function CustomCheckModal({
+  open = false,
+  selectedCheckID,
+  selectedCheckValues,
+  selectedCheckDescription,
+  provider = 'Unknown',
+  isChecked = false,
+  onClose = noop,
+  onSave = noop,
+}) {
+  const [checked, setChecked] = useState(isChecked);
+  const [customValues, setCustomValues] = useState({});
+
+  const checkBoxWarningText =
+    'Trento & SUSE cannot be held liable for damages if system is unable to function due to custom check value.';
+  const checkTitle = `Check: ${selectedCheckID}`;
+
+  const handleCustomValueInput = (name, value) => {
+    setCustomValues((previousValues) => ({
+      ...previousValues,
+      [name]: value,
+    }));
+  };
+
+  const buildCustomCheckPayload = (checkID, values) => {
+    const payload = {
+      checksID: checkID,
+      customValues: values,
+    };
+    return payload;
+  };
+  const renderLabelWithTooltip = (name) => {
+    const labelContent = (
+      <Label className="block truncate max-w-[200px] sm:max-w-[250px] md:max-w-[300px]">
+        {name}:
+      </Label>
+    );
+
+    return name?.length > 25 ? (
+      <Tooltip zIndex="50" content={name}>
+        {labelContent}
+      </Tooltip>
+    ) : (
+      labelContent
+    );
+  };
+
+  return (
+    <Modal
+      className="!w-3/4 !max-w-3xl"
+      title={checkTitle}
+      open={open}
+      onClose={() => {
+        setCustomValues({});
+        setChecked(false);
+        onClose();
+      }}
+    >
+      <p className="text-gray-500 text-sm font-normal tracking-wide pb-2">
+        {selectedCheckDescription}
+      </p>
+
+      <div className="flex items-center border border-yellow-400 bg-yellow-50 p-4 rounded-md text-yellow-600 mb-4">
+        <Input
+          type="checkbox"
+          checked={checked}
+          onChange={() => setChecked((prev) => !prev)}
+        />
+        <EOS_WARNING_OUTLINED
+          size="xxl"
+          className="centered fill-yellow-500 ml-4 mr-4"
+        />
+        <span className="font-semibold">{checkBoxWarningText}</span>
+      </div>
+      {selectedCheckValues?.map((value) => (
+        <div
+          key={`${value?.name}_${value?.default}`}
+          className="flex items-center space-x-2 mb-8"
+        >
+          <div className="flex-col w-1/3 min-w-[200px]">
+            {renderLabelWithTooltip(value?.name)}
```

Comment:
This function allows us to add ellipsis(…) and a tooltip when the checks value becomes to long. 
With this the labels and input fields stay aligned